plugin/someteam.example.com/v1/gogetter/GoGetter

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+# Copyright 2019 The Kubernetes Authors.
+# SPDX-License-Identifier: Apache-2.0
+
+# HashiCorp go-getter
+#
+# Reads a file like this
+#
+#  apiVersion: someteam.example.com/v1
+#  kind: GoGetter
+#  metadata:
+#    name: example
+#  url: github.com/kustless/kustomize-examples.git
+#  # subPath: base # (optional) relative path in the package
+#  # command: cat *.yaml # (optional) build command, default `kustomize build $subPath`
```

Comment:
The `command` can be optional. When the url contains a `kustomization.yaml`, this plugin should be able to auto check it and trigger kustomize build on it. When the url doesn't contain a `kustomization.yaml`, this plugin can load all the yaml files that have `apiVersion` and `kind`.

Reply:
`auto check` might be overkilled for a simple shell script.

Consider the use case of downloading ksonnet or helmfile configuration or selectively apply part of the downloaded manifests.